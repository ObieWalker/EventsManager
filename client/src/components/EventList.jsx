import React from 'react';
import PropTypes from 'prop-types';
import { Col } from 'react-materialize';

const EventCard = (props) => {
  const { event } = props;
  if (!event.isCancelled) {
    return (
      <div style={{ margin: '5%' }}>
        <Col s={10} m={4}>
          <div className="col 12">
            <div
              className="card elegant-color white-text"
              style={{ borderRadius: '10%', width: '18rem', padding: '5%' }}
            >
              <div className="card-body">
                <h5 className="card-title white-text">{event.eventType}</h5>
                <p className="card-text white-text">
                  Address: {event.Center.address}, {event.Center.city}
                </p>
                <p className="card-text white-text">Date: {event.date}</p>
                <p className="card-text white-text">
                  Guest Estimate: {event.guestNo}
                </p>
                <hr className="hr-light" />
              </div>
            </div>
          </div>
        </Col>
      </div>
    );
  }
  return null;
};

EventCard.propTypes = {
  event: PropTypes.object,
  center: PropTypes.object,
  handleShowModal: PropTypes.func,
  name: PropTypes.string,
  address: PropTypes.string,
  city: PropTypes.string,
  facility: PropTypes.string,
  getAllCenters: PropTypes.func,
  getCenterEvents: PropTypes.func
};

export default EventCard;
