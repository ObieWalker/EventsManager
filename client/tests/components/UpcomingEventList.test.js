import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import EventList from '../../src/components/EventList.jsx';

let mountedForm;
let props;
let wrapper;

const getComponent = () => {
  if (!mountedForm) {
    props = {
      event: {
        Center: {}
      },
      center: {},
      handleShowModal: jest.fn(),
      name: '',
      address: '',
      city: '',
      facility: '',
      getAllCenters: jest.fn(),
      getCenterEvents: jest.fn()
    };
    mountedForm = shallow(<EventList {...props} />);
  }
  return mountedForm;
};

describe('Event card for upcoming events Component', () => {
  beforeEach(() => {});

  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });
  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper.find('Col').length).toBe(1);
    expect(wrapper.find('div').length).toBe(4);
    expect(wrapper.find('p').length).toBe(3);
    expect(wrapper.find('h5').length).toBe(1);
  });
});
