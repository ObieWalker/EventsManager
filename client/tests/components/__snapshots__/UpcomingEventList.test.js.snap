// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event card for upcoming events Component renders component successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventCard
    address=""
    center={Object {}}
    city=""
    event={
      Object {
        "Center": Object {},
      }
    }
    facility=""
    getAllCenters={[MockFunction]}
    getCenterEvents={[MockFunction]}
    handleShowModal={[MockFunction]}
    name=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Col
        m={4}
        s={10}
      >
        <div
          className="col 12"
        >
          <div
            className="card elegant-color white-text"
            style={
              Object {
                "borderRadius": "10%",
                "padding": "5%",
                "width": "18rem",
              }
            }
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title white-text"
              />
              <p
                className="card-text white-text"
              >
                Address: 
                , 
              </p>
              <p
                className="card-text white-text"
              >
                Date: 
              </p>
              <p
                className="card-text white-text"
              >
                Guest Estimate: 
              </p>
              <hr
                className="hr-light"
              />
            </div>
          </div>
        </div>
      </Col>,
      "style": Object {
        "display": "inlineBlock",
        "float": "left",
        "margin": "1%",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="col 12"
        >
          <div
            className="card elegant-color white-text"
            style={
              Object {
                "borderRadius": "10%",
                "padding": "5%",
                "width": "18rem",
              }
            }
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title white-text"
              />
              <p
                className="card-text white-text"
              >
                Address: 
                , 
              </p>
              <p
                className="card-text white-text"
              >
                Date: 
              </p>
              <p
                className="card-text white-text"
              >
                Guest Estimate: 
              </p>
              <hr
                className="hr-light"
              />
            </div>
          </div>
        </div>,
        "m": 4,
        "s": 10,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="card elegant-color white-text"
            style={
              Object {
                "borderRadius": "10%",
                "padding": "5%",
                "width": "18rem",
              }
            }
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title white-text"
              />
              <p
                className="card-text white-text"
              >
                Address: 
                , 
              </p>
              <p
                className="card-text white-text"
              >
                Date: 
              </p>
              <p
                className="card-text white-text"
              >
                Guest Estimate: 
              </p>
              <hr
                className="hr-light"
              />
            </div>
          </div>,
          "className": "col 12",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="card-body"
            >
              <h5
                className="card-title white-text"
              />
              <p
                className="card-text white-text"
              >
                Address: 
                , 
              </p>
              <p
                className="card-text white-text"
              >
                Date: 
              </p>
              <p
                className="card-text white-text"
              >
                Guest Estimate: 
              </p>
              <hr
                className="hr-light"
              />
            </div>,
            "className": "card elegant-color white-text",
            "style": Object {
              "borderRadius": "10%",
              "padding": "5%",
              "width": "18rem",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h5
                  className="card-title white-text"
                />,
                <p
                  className="card-text white-text"
                >
                  Address: 
                  , 
                </p>,
                <p
                  className="card-text white-text"
                >
                  Date: 
                </p>,
                <p
                  className="card-text white-text"
                >
                  Guest Estimate: 
                </p>,
                <hr
                  className="hr-light"
                />,
              ],
              "className": "card-body",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "card-title white-text",
                },
                "ref": null,
                "rendered": null,
                "type": "h5",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Address: ",
                    undefined,
                    ", ",
                    undefined,
                  ],
                  "className": "card-text white-text",
                },
                "ref": null,
                "rendered": Array [
                  "Address: ",
                  undefined,
                  ", ",
                  undefined,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Date: ",
                    undefined,
                  ],
                  "className": "card-text white-text",
                },
                "ref": null,
                "rendered": Array [
                  "Date: ",
                  undefined,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Guest Estimate: ",
                    undefined,
                  ],
                  "className": "card-text white-text",
                },
                "ref": null,
                "rendered": Array [
                  "Guest Estimate: ",
                  undefined,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "hr-light",
                },
                "ref": null,
                "rendered": null,
                "type": "hr",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Col
          m={4}
          s={10}
        >
          <div
            className="col 12"
          >
            <div
              className="card elegant-color white-text"
              style={
                Object {
                  "borderRadius": "10%",
                  "padding": "5%",
                  "width": "18rem",
                }
              }
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title white-text"
                />
                <p
                  className="card-text white-text"
                >
                  Address: 
                  , 
                </p>
                <p
                  className="card-text white-text"
                >
                  Date: 
                </p>
                <p
                  className="card-text white-text"
                >
                  Guest Estimate: 
                </p>
                <hr
                  className="hr-light"
                />
              </div>
            </div>
          </div>
        </Col>,
        "style": Object {
          "display": "inlineBlock",
          "float": "left",
          "margin": "1%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <div
            className="col 12"
          >
            <div
              className="card elegant-color white-text"
              style={
                Object {
                  "borderRadius": "10%",
                  "padding": "5%",
                  "width": "18rem",
                }
              }
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title white-text"
                />
                <p
                  className="card-text white-text"
                >
                  Address: 
                  , 
                </p>
                <p
                  className="card-text white-text"
                >
                  Date: 
                </p>
                <p
                  className="card-text white-text"
                >
                  Guest Estimate: 
                </p>
                <hr
                  className="hr-light"
                />
              </div>
            </div>
          </div>,
          "m": 4,
          "s": 10,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="card elegant-color white-text"
              style={
                Object {
                  "borderRadius": "10%",
                  "padding": "5%",
                  "width": "18rem",
                }
              }
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title white-text"
                />
                <p
                  className="card-text white-text"
                >
                  Address: 
                  , 
                </p>
                <p
                  className="card-text white-text"
                >
                  Date: 
                </p>
                <p
                  className="card-text white-text"
                >
                  Guest Estimate: 
                </p>
                <hr
                  className="hr-light"
                />
              </div>
            </div>,
            "className": "col 12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="card-body"
              >
                <h5
                  className="card-title white-text"
                />
                <p
                  className="card-text white-text"
                >
                  Address: 
                  , 
                </p>
                <p
                  className="card-text white-text"
                >
                  Date: 
                </p>
                <p
                  className="card-text white-text"
                >
                  Guest Estimate: 
                </p>
                <hr
                  className="hr-light"
                />
              </div>,
              "className": "card elegant-color white-text",
              "style": Object {
                "borderRadius": "10%",
                "padding": "5%",
                "width": "18rem",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h5
                    className="card-title white-text"
                  />,
                  <p
                    className="card-text white-text"
                  >
                    Address: 
                    , 
                  </p>,
                  <p
                    className="card-text white-text"
                  >
                    Date: 
                  </p>,
                  <p
                    className="card-text white-text"
                  >
                    Guest Estimate: 
                  </p>,
                  <hr
                    className="hr-light"
                  />,
                ],
                "className": "card-body",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "card-title white-text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h5",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Address: ",
                      undefined,
                      ", ",
                      undefined,
                    ],
                    "className": "card-text white-text",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Address: ",
                    undefined,
                    ", ",
                    undefined,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Date: ",
                      undefined,
                    ],
                    "className": "card-text white-text",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Date: ",
                    undefined,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Guest Estimate: ",
                      undefined,
                    ],
                    "className": "card-text white-text",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Guest Estimate: ",
                    undefined,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "hr-light",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "hr",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
